### PR TITLE
Pgsql 修复查询报错，实例为null报错

### PR DESCRIPTION
**Pull request by HJY-dev**

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs [hunk 1]
```diff
@@ -147,7 +147,7 @@ public async Task<IActionResult> Index(WorkflowTypeIndexOptions options, PagerPa
                     {
                         WorkflowType = x,
                         Id = x.Id,
-                        HasInstances = workflowTypeIdsWithInstances.Contains(x.WorkflowTypeId),
+                        HasInstances = workflowTypeIdsWithInstances == null ? false : workflowTypeIdsWithInstances.Contains(x.WorkflowTypeId),
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs:150]: Simplify it to 

```
workflowTypeIdsWithInstances != null && workflowTypeIdsWithInstances.Contains(x.WorkflowTypeId)
```

File: src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs [hunk 2]
```diff
@@ -128,7 +128,7 @@ public async Task<IActionResult> Index(WorkflowTypeIndexOptions options, PagerPa
             var sqlBuilder = dialect.CreateBuilder(_session.Store.Configuration.TablePrefix);
             sqlBuilder.Select();
             sqlBuilder.Distinct();
-            sqlBuilder.Selector(nameof(WorkflowIndex.WorkflowTypeId));
+            sqlBuilder.Selector(nameof(WorkflowIndex), nameof(WorkflowIndex.WorkflowTypeId), _session.Store.Configuration.Schema);
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs:131]: Good catch! There are 4 other places in OC where with use `Selector` without providing a schema. `DbConnectionValidator` (2 times), `IndexingTaskManager`, and `SqlParser`. Can you please update these files also?

Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs:131]: Why do you need this? Why can't you use the `Selector` overload without the column and the schema?

File: src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs (hunk 1 above)
Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs:150]: Maybe the following but can it be null or just empty?

    workflowTypeIdsWithInstances?.Contains(x.WorkflowTypeId) ?? false,

File: src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs (hunk 2 above)
Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs:131]: Yes, sould be checked / tried ;)

Comment (HJY-dev) [reply src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs:131]: 修改前 sqlBuilder.Selector(nameof(WorkflowIndex));
生成的sql语句：SELECT DISTINCT WorkflowIndex FROM "WorkflowIndex"

修改后 sqlBuilder.Selector(nameof(WorkflowIndex), nameof(WorkflowIndex.WorkflowTypeId), _session.Store.Configuration.Schema); 
生成的sql语句：SELECT DISTINCT "WorkflowIndex"."WorkflowTypeId" FROM "WorkflowIndex"

File: src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs (hunk 1 above)
Comment (HJY-dev) [reply src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs:150]: 如果流程没有触发，不会产生流程实例，此处会报错，没判断是否为null。

File: src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs (hunk 2 above)
Comment (HJY-dev) [reply src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs:131]: a51424aa190ff4946cc293010b7311a

File: src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs (hunk 1 above)
Comment (hyzx86) [reply src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs:150]: @HJY-dev  更新 你的代码 成这样 就好了  `workflowTypeIdsWithInstances?.Contains(x.WorkflowTypeId) ?? false,`

File: src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs (hunk 2 above)
Comment (HJY-dev) [reply src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs:131]: image

**Pull request closed without merging**